libgit2-sys/lib.rs

```diff
@@ -600,6 +603,7 @@ pub enum git_ref_t {
 }
 pub use git_ref_t::*;
 
+enum_from_primitive!{
```

Comment:
Do we know that `git_filemode_t` is layout-compatible with an `i32`?  It looks like C could choose a representation using either `i32` or `u16`.  (Although it looks like anything `rustc` currently supports will use `i32`, so we're probably OK.)